### ISSUE TITLE
Current release of dependency pkg/errors breaks things

### ISSUE THREAD

**Issue by ssgelm** [git-lfs/git-lfs#2390]
As part of getting git-lfs included with debian I am working on getting the package to build without any vendored dependencies.  This has been mostly ok, but I am running into a bug created by https://github.com/pkg/errors/commits/777ed74de5b8cdca2147e5da066c3bd418c26083.  This causes the following test faulire in git-lfs:
```
--- FAIL: TestBehaviorWraps (0.00s)
        errors_test.go:37: expected wrapped error to also be fatal
```
I'm working on tracking down the bug but I figured I'd file an issue in case someone with a deeper knowledge of the code had any ideas.  cc @technoweenie @ttaylorr

**Comment by ttaylorr** [comment]
Hi @ssgelm, thanks for opening this issue. I was able to reproduce the issue locally, and I have a vague idea of what's going on and causing it.

I'm commenting to ack this issue internally, but want to wait on hearing back about https://github.com/git-lfs/git-lfs/issues/2354#issuecomment-313525487 before moving forward and trying to address this (though I think we should eventually, either way).

**Comment by ssgelm** [comment]
@ttaylorr Soungs good.  I'll see what I can do myself as well to get it working again.

**Comment by ssgelm** [comment]
I'll send a proper PR but I figured out the problem.  In the newer version upstream errors end up doubly-wrapped.  My fix is below.

```diff
diff --git a/errors/types.go b/errors/types.go
index efeca36c..d1f431fd 100644
--- a/errors/types.go
+++ b/errors/types.go
@@ -336,8 +336,14 @@ func NewRetriableError(err error) error {
 }

 func parentOf(err error) error {
-       if c, ok := err.(errorWithCause); ok {
-               return c.Cause()
+  type causer interface {
+    Cause() error
+  }
+
+       if c, ok := err.(causer); ok {
+    if innerC, innerOk := c.Cause().(causer); innerOk {
+      return innerC.Cause()
+    }
        }

        return nil
```

**Comment by ttaylorr** [comment]
@ssgelm nice find, and thanks for looking into this.

> In the newer version upstream errors end up doubly-wrapped.

Do you think this is a bug introduced upstream? I wonder if we should investigate the cause there, and send a PR to remove this double-wrapping behavior.

Either way, the diff you posted shouldn't change anything from an LFS perspective, since the `parentOf()` method is what drives all of the `errors.Is<t>Error` methods.

**Comment by ssgelm** [comment]
@ttaylorr Great question.  Given the way they have written the code I don't think this is a bug for them per se.  Their `Cause` function traverses all the way up to the root Cause so while it is theoretically _slightly_ slower I don't think it changes their API.  I spent a fair amount of time looking at their code and I don't think the fix is very simple - the commit that broke things here basically added another layer.  Unless you feel strongly otherwise my vote would be to upgrade to the latest version of `pkg/errors` and include my fix.  I'll PR that today.

**Comment by ssgelm** [comment]
Yeah reading more about their rationale at https://github.com/pkg/errors/releases/tag/v0.8.0 I believe it was an intentional change.

**Comment by ssgelm** [comment]
This is now fixed with #2426
